### PR TITLE
F/deficon rn

### PR DESCRIPTION
**Pull request by matt-7t**

### CODE REVIEW

File: src/fontawesome/react_native.cljc [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+(ns fontawesome.react-native
+  #?(:cljs (:require-macros [fontawesome.react-native]))
+  (:require [helix.core]
+            #?@(:cljs [[react-native]])))
+
+#?(:clj
+   (defmacro deficon
+     [icon-name filepath]
+     (let [props (gensym)]
+       `(let [svg-comp# (.-default (js/require ~filepath))]
```
Comment (matt-7t) [inline src/fontawesome/react_native.cljc:19]: Would like to know if:
 1) top-level `let` providing scoped bindings to `def` calls is bad practice or not
 2) gensym vs auto-gensym? should I prefer one over the other?

Comment (tom7t) [reply src/fontawesome/react_native.cljc:19]: 1. I think the let you have here is pretty normal practice with a macro, as the generated code is just a `defnc` once the macro is done.
2. I would use the `auto-gensym` in macros. Any particular reason you've done `gensym` only for the props? Haven't done a macro for a while so might be missing something.

File: src/fontawesome/react_native.cljc [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+          (helix.core/defnc ~icon-name
+            [~props]
+            {:helix/features {:fast-refresh true}
+             :wrap [(helix.core/memo =)]}
+            (let [style# (get ~props :style)
+                  svg-props# (merge (dissoc ~props style#)
+                                    (when-let [fill# (.-color style#)]
+                                      {:fill fill#}))]
+              (helix.core/$ react-native/View {:style style#}
```
Comment (matt-7t) [inline src/fontawesome/react_native.cljc:28]: Didn't think `react-native/View` would work like this but it seems to (including in release), looking forward to being burned by it. Haha

File: src/fontawesome/react_native.cljc (hunk 1 above)
Comment (matt-7t) [reply src/fontawesome/react_native.cljc]: gensym'd props was like that before, converted to autosym now haha

**Review by sandhu — APPROVED** [review]

**Pull request merged**